pkg/process/ptrace/cwrapper_linux.go

```diff
@@ -0,0 +1,29 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build cgo
```

Comment:
Do we need to update the [Makefile](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/87774e00dd509a1bf7a22678f799756706be6755/Makefile#L33) to enable cgo for this? Trying to build as is I got:

```
CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -o otel-go-instrumentation cli/main.go
# github.com/open-telemetry/opentelemetry-go-instrumentation/pkg/process/ptrace
pkg/process/ptrace/ptrace_linux.go:53:9: undefined: waitpid
make: *** [Makefile:33: build] Error 1
```

Reply:
And enabling cgo gives this error at runtime:
```
/otel-go-instrumentation: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.32' not found (required by /otel-go-instrumentation)
/otel-go-instrumentation: /lib/x86_64-linux-gnu/libc.so.6: version `GLIBC_2.34' not found (required by /otel-go-instrumentation)
```

Tried adding `glibc` to the [Dockerfile](https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/87774e00dd509a1bf7a22678f799756706be6755/Dockerfile#L2) but got the same result. Maybe a difference between the fedora:37 base image and the distroless image?

Reply:
I changed the base image to `registry.fedoraproject.org/fedora-minimal:35` which includes glibc and works fine. The image can probably be smaller, but I think it is good enough for the first release.